src/future.rs

```diff
@@ -42,8 +42,26 @@ impl<T: Generator<Yield = ()>> Future for GenFuture<T> {
     }
 }
 
-#[thread_local]
+#[cfg_attr(not(feature = "single-thread"), thread_local)]
```

Comment:
This change to a cfg_attr is unnecessary, the attr will only get applied if the item isn't cfg'd out.

Reply:
Huh, you're right. That's much smarter behavior that my mental model of attributes lead me to expect. Reverted this line.